File: Application_part/project_final_version/app/src/main/java/com/example/heartbeat/CSVFile.java
package com.example.heartbeat;

import android.os.Environment;
import android.util.Log;

import com.example.heartbeat.ECG.ECGData;
import com.example.heartbeat.PPG.PPGData;
import com.example.heartbeat.Temperature.TempData;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class CSVFile {
    String fileTitle;

    public CSVFile(String filename){
        this.fileTitle = filename + ".csv";
        File file = new File(Environment.getExternalStorageDirectory(), fileTitle);
        try{
            FileWriter writer = new FileWriter(file, false);
            if(!file.exists()) {
                file.createNewFile();
            }

            if(filename.equals("Temp")) {
                writer.append("Count");
                writer.append(",");
                writer.append("Temperature");
                writer.append("\n");
            }
            if(filename.equals("PPG")){
                writer.append("Count");
                writer.append(",");
                writer.append("grnCnt");
                writer.append(",");
                writer.append("grn2Cnt");
                writer.append(",");
                writer.append("hr");
                writer.append("\n");
            }
            if(filename.equals("ECG")){
                writer.append("Count");
                writer.append(",");
                writer.append("RAW ECG1");
                writer.append(",");
                writer.append("RAW ECG2");
                writer.append(",");
                writer.append("RAW ECG3");
                writer.append(",");
                writer.append("RAW ECG4");
                writer.append(",");
                writer.append("eTAg1");
                writer.append(",");
                writer.append("eTAG2");
                writer.append(",");
                writer.append("eTAG3");
                writer.append(",");
                writer.append("eTAG4");
                writer.append("\n");
            }

            writer.close();

        }catch(IOException e){
            e.printStackTrace();
        }
    }

    void writeTempFile(TempData tempData){
        File file = new File(Environment.getExternalStorageDirectory(), this.fileTitle);

        try{
            FileWriter writer = new FileWriter(file, true);
            if(!file.exists()) {
                file.createNewFile();
            }
            writer.append(Integer.toString(tempData.getCount()));
            writer.append(",");
            writer.append(Double.toString(tempData.getValue()));
            writer.append("\n");
            writer.close();

        }catch(IOException e){
            e.printStackTrace();
        }
    }

    void writePPGFile(PPGData ppgData){
        File file = new File(Environment.getExternalStorageDirectory(), this.fileTitle);

        try{
            FileWriter writer = new FileWriter(file, true);
            if(!file.exists()) {
                file.createNewFile();
            }
            writer.append(Integer.toString(ppgData.getCount()));
            writer.append(",");
            writer.append(Double.toString(ppgData.getGrnCnt()));
            writer.append(",");
            writer.append(Double.toString(ppgData.getGrn2Cnt()));
            writer.append(",");
            writer.append(Integer.toString(ppgData.getHeartRate()));
            writer.append("\n");
            writer.close();

        }catch(IOException e){
            e.printStackTrace();
        }
    }

    void writeECGFile(ECGData ecgData){
        File file = new File(Environment.getExternalStorageDirectory(), this.fileTitle);

        try{
            FileWriter writer = new FileWriter(file, true);
            if(!file.exists()) {
                file.createNewFile();
            }
            writer.append(Integer.toString(ecgData.getCount()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg1()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg2()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg3()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg4()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg1eTag()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg2eTag()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg3eTag()));
            writer.append(",");
            writer.append(Integer.toString(ecgData.getEcg4eTag()));
            writer.append("\n");
            writer.close();

        }catch(IOException e){
            e.printStackTrace();
        }
    }

}

File: Application_part/project_kotlin_version/Kotlin_ver/app/src/main/java/com/example/kotlin_ver/CSVFile.kt
package com.example.kotlin_ver

import android.os.Environment
import com.example.kotlin_ver.ECG.ECGData
import com.example.kotlin_ver.PPG.PPGData
import com.example.kotlin_ver.Temperature.TempData
import java.io.File
import java.io.FileWriter
import java.io.IOException

class CSVFile {
    var fileTitle: String? = null

    constructor(filename: String) {
        fileTitle = "$filename.csv"
        val file = File(Environment.getExternalStorageDirectory(), fileTitle)
        try {
            val writer = FileWriter(file, false)
            if (!file.exists()) {
                file.createNewFile()
            }
            if (filename == "Temp") {
                writer.append("Count")
                writer.append(",")
                writer.append("Temperature")
                writer.append("\n")
            }
            if (filename == "PPG") {
                writer.append("Count")
                writer.append(",")
                writer.append("grnCnt")
                writer.append(",")
                writer.append("grn2Cnt")
                writer.append(",")
                writer.append("hr")
                writer.append("\n")
            }
            if (filename == "ECG") {
                writer.append("Count")
                writer.append(",")
                writer.append("RAW ECG1")
                writer.append(",")
                writer.append("RAW ECG2")
                writer.append(",")
                writer.append("RAW ECG3")
                writer.append(",")
                writer.append("RAW ECG4")
                writer.append(",")
                writer.append("eTAg1")
                writer.append(",")
                writer.append("eTAG2")
                writer.append(",")
                writer.append("eTAG3")
                writer.append(",")
                writer.append("eTAG4")
                writer.append("\n")
            }
            writer.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun writeTempFile(tempData: TempData) {
        val file = File(Environment.getExternalStorageDirectory(), fileTitle)
        try {
            val writer = FileWriter(file, true)
            if (!file.exists()) {
                file.createNewFile()
            }
            writer.append("${tempData.count}")
            writer.append(",")
            writer.append("${tempData.value}")
            writer.append("\n")
            writer.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun writePPGFile(ppgData: PPGData) {
        val file = File(Environment.getExternalStorageDirectory(), fileTitle)
        try {
            val writer = FileWriter(file, true)
            if (!file.exists()) {
                file.createNewFile()
            }
            writer.append("${ppgData.count}")
            writer.append(",")
            writer.append("${ppgData.grnCnt}")
            writer.append(",")
            writer.append("${ppgData.grn2Cnt}")
            writer.append(",")
            writer.append("${ppgData.heartRate}")
            writer.append("\n")
            writer.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun writeECGFile(ecgData: ECGData) {
        val file = File(Environment.getExternalStorageDirectory(), fileTitle)
        try {
            val writer = FileWriter(file, true)
            if (!file.exists()) {
                file.createNewFile()
            }
            writer.append("${ecgData.count}")
            writer.append(",")
            writer.append("${ecgData.ecg1}")
            writer.append(",")
            writer.append("${ecgData.ecg2}")
            writer.append(",")
            writer.append("${ecgData.ecg3}")
            writer.append(",")
            writer.append("${ecgData.ecg4}")
            writer.append(",")
            writer.append("${ecgData.ecg1eTag}")
            writer.append(",")
            writer.append("${ecgData.ecg2eTag}")
            writer.append(",")
            writer.append("${ecgData.ecg3eTag}")
            writer.append(",")
            writer.append("${ecgData.ecg4eTag}")
            writer.append("\n")
            writer.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}